### PR TITLE
bl09xx driver for both bl0939 and bl0940

### PR DESCRIPTION
**Pull request by barbudor**
## Description:

Following release of Sonoff Dual R3 with BL0939 energy chip
Improved BL0940 driver (XNRG_14) to support both BL0939 and BL0940
A different RX pin should be assigned to distinguish between the 2 : `BL0939 RX` and `BL0940 RX`
The TX pin should be `Serial TX` in any case.

https://github.com/arendst/Tasmota/discussions/10793#discussioncomment-1335467

New `USE_BL09XX` replace `USE_BL0940` but the driver will still be included if `USE_BL0940` is used by legacy `user_confg_override.h` with a warning stating it is obsolete.
The `(+1k6 code)` in the comment of USE_BL094XX hasn't been verified for accuracy

Only tested on ESP32 for Sonoff Dual R3 v2
Should be counter-tested on Blitzwolf SHP6 to check still compliant with BL0940 (I don't have the device)

## Checklist:
  - [X] The pull request is done against the latest development branch
  - [X] Only relevant files were touched
  - [X] Only one feature/fix was added per PR and the code change compiles without warnings
  - [ ] The code change is tested and works with Tasmota core ESP8266 V.2.7.4.9
  - [X] The code change is tested and works with Tasmota core ESP32 V.1.0.7.4
  - [X] I accept the [CLA](https://github.com/arendst/Tasmota/blob/development/CONTRIBUTING.md#contributor-license-agreement-cla).

_NOTE: The code change must pass CI tests. **Your PR cannot be merged unless tests pass**_

### CODE REVIEW

File: tasmota/xnrg_14_bl09xx.ino [hunk 1]
```diff
@@ -0,0 +1,388 @@
...
+
+  You should have received a copy of the GNU General Public License
+  along with this program.  If not, see <http://www.gnu.org/licenses/>.
+*/
+
+#ifdef USE_ENERGY_SENSOR
+#if defined(USE_BL0940) || defined(USE_BL09XX)
+#ifdef USE_BL0940
+#warning **** USE_BL0940 is obsolete. Please replace with USE_BLE0939 ****
+#endif
```
[the remark this answers is not in the document]
Comment (barbudor) [reply tasmota/xnrg_14_bl09xx.ino:24]: fixed

**Comment by arendst** [comment]
Did some testing on the SHP10

The changed default referrnces make the SHP10 calibration way off. Also the current formula has changed resulting on a deviation of 100.

I'll merge this and will upload an updated driver solving both issues.

**Comment by barbudor** [comment]
Hi Theo,
That's strange because the formula are the same in both datasheets. I suspect the 2 chips to be based on the same die with less pins exposed for the 0940.
Have you cleared the old calibration in your plug?

**Comment by arendst** [comment]
The formula might be the same; the shunt resistance probably isn't. 

To keep legacy support a user shouldn't need re-calibration with this driver.

To satisfy you ;-) I'll re-init the default values with your suggested values and report (suspected mis-) values...

**Comment by arendst** [comment]
As expected: 

image

Values twice the real value due to different shunt resistor used in shp10/dual r3.

For now let's keep the driver as it is.

**Comment by barbudor** [comment]
Thanks for checking

**Pull request merged**